plugins/module_utils/utils.py

```diff
@@ -264,6 +297,51 @@ def op(operation, path, value):
     return {"op": operation, "path": path, "value": value}
 
 
+def safe_json(data: str) -> Dict:
+    try:
+        json_object = json.loads(data)
+    except (ValueError, TypeError) as e:
+        return data
+    return json_object
+
+
+def merge_dicts(list1: List, list2) -> List:
```

Comment:
What's the problem that this function is trying to solve? Is there an example of a resource property that needs this? It makes some assumptions about the structure of the arguments that at the very least should probably be documented. I'm worried that those assumptions may not always be true. For example, the following will raise an exception:

```python
old = [
    {"foo": "bar"}
]
new = [
    {"foo": "gaz", "someotherkey": {"blah": "asdf"}}
]
merge_dicts(old, new)
```

Reply:
@gravesm I fixed that corner case and add a test case for it. However, this function handles the case when two list of dictionaries (with keys different than Key) needs to merged. For example:

- I create in a first instance the role below and then try to re-run the same task (because I want to check idempotency of update the role) it will fail because it will be unable to compare list of dicts inside `policies` parameter.
```
    - name: Create minimal lambda role
      amazon.cloud.iam_role:
        role_name: '{{ lambda_role_name }}'
        assume_role_policy_document: '{{ lookup("file", "lambda-minimal-policy.json") }}'
        policies:
          - policy_name: "{{ lambda_policy_name }}"
            policy_document: '{{ lookup("file", "lambda-policy.json") }}'
        managed_policy_arns:
          - "arn:aws:iam::{{ _caller_info.account }}:policy/{{ lambda_policy_name }}"
        wait: true
      register: _create_iam_role_result 
```

Also, policy_document returned by the API will be a dictionary while the one the user will specify will be a string, so, that's why I created safe_load().

We should also ensure the right quotes are used within the policy_document and that's why I had to add `QuoteSwappingEncoder`. The test case I added covers exactly the case where we need this quotes swapping.